### PR TITLE
Fix mail parsing in collector

### LINKED ISSUE #7643: [9.5] New IMAP Library Does Not Handle No Reply-To in Email Headers
**Describe the bug**

Since upgrading our system to 9.5.0, we have discovered that our email receiver does not work as expected. On further investigation, it appears that it now cannot handle emails in which we do not have a Reply-To header.

This is the error we get when trying to receive emails.

```
[2020-07-09 16:45:30] glpiphplog.CRITICAL:   *** Uncaught Exception Laminas\Mail\Storage\Exception\InvalidArgumentException: Header with Name reply-to or reply-to not found in /var/www/html/glpi/vendor/laminas/laminas-mail/src/Storage/Part.php at line 307
  Backtrace :
  inc/mailcollector.class.php:955                    Laminas\Mail\Storage\Part->getHeader()
  inc/mailcollector.class.php:711                    MailCollector->buildTicket()
  front/mailcollector.form.php:77                    MailCollector->collect()
```

It should be noted that we do have an In-Reply-To header. I would also like to note that it doesn't seem to handle emails where there is no To header (for example, all recipients are in the BCC header).

**Page(s) URL**

https://{GLPIHOST}/glpi/front/mailcollector.form.php

**To reproduce**

Set up a mail receiver.
Have an email with either no To header, or not Reply-To header in the inbox.
Attempt to retrieve the email.

**Expected behavior**

The email should be imported into GLPI.

**Logs**

```
[2020-07-09 16:45:30] glpiphplog.CRITICAL:   *** Uncaught Exception Laminas\Mail\Storage\Exception\InvalidArgumentException: Header with Name reply-to or reply-to not found in /var/www/html/glpi/vendor/laminas/laminas-mail/src/Storage/Part.php at line 307
  Backtrace :
  inc/mailcollector.class.php:955                    Laminas\Mail\Storage\Part->getHeader()
  inc/mailcollector.class.php:711                    MailCollector->buildTicket()
  front/mailcollector.form.php:77                    MailCollector->collect()  
```

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```
**General**
Operating system: Linux itsm01 18.04.1-Ubuntu SMP Tue Jun 30 23:04:16 UTC 2020 x86_64
PHP 7.2.24-0ubuntu apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apc,
	apcu, calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, intl, json, ldap, libxml,
	mbstring, mysqli, mysqlnd, openssl, pcre, pdo_mysql, posix, readline, session, shmop, sockets, sodium, standard, sysvmsg,
	sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zlib)
Setup: max_execution_time="600" memory_limit="256M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="50M" 
Software: Apache/2.4.29 (Ubuntu) (Apache/2.4.29 (Ubuntu) Server at REDACTED Port 443)
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36
Server Software: (Ubuntu)
	Server Version: ubuntu
	Server SQL Mode: 
	Parameters: glpiuser@localhost/glpi
	Host info: Localhost via UNIX socket
	
PHP version is at least 7.2.0 - Perfect!PHP version is at least 7.2.0 - Perfect!
Sessions support is available - Perfect!Sessions support is available - Perfect!
Allocated memory > 64 Mio - Perfect!Allocated memory > 64 Mio - Perfect!
mysqli extension is installedmysqli extension is installed
ctype extension is installedctype extension is installed
fileinfo extension is installedfileinfo extension is installed
json extension is installedjson extension is installed
mbstring extension is installedmbstring extension is installed
iconv extension is installediconv extension is installed
zlib extension is installedzlib extension is installed
curl extension is installedcurl extension is installed
gd extension is installedgd extension is installed
simplexml extension is installedsimplexml extension is installed
intl extension is installedintl extension is installed
ldap extension is installedldap extension is installed
apcu extension is installedapcu extension is installed
Zend OPcache extension is installedZend OPcache extension is installed
xmlrpc extension is installedxmlrpc extension is installed
CAS extension is installedCAS extension is installed
exif extension is installedexif extension is installed
zip extension is not presentzip extension is not present
bz2 extension is not presentbz2 extension is not present
sodium extension is installedsodium extension is installed
Database version seems correct (5.7.30) - Perfect!Database version seems correct (5.7.30) - Perfect!
Timezones seems loaded in databaseTimezones seems loaded in database
The log file has been created successfully.The log file has been created successfully.
Write access to /etc/glpi/ has been validated.Write access to /etc/glpi/ has been validated.
Write access to /var/lib/glpi has been validated.Write access to /var/lib/glpi has been validated.
Write access to /var/lib/glpi/_dumps has been validated.Write access to /var/lib/glpi/_dumps has been validated.
Write access to /var/lib/glpi/_sessions has been validated.Write access to /var/lib/glpi/_sessions has been validated.
Write access to /var/lib/glpi/_cron has been validated.Write access to /var/lib/glpi/_cron has been validated.
Write access to /var/lib/glpi/_graphs has been validated.Write access to /var/lib/glpi/_graphs has been validated.
Write access to /var/lib/glpi/_lock has been validated.Write access to /var/lib/glpi/_lock has been validated.
Write access to /var/lib/glpi/_plugins has been validated.Write access to /var/lib/glpi/_plugins has been validated.
Write access to /var/lib/glpi/_tmp has been validated.Write access to /var/lib/glpi/_tmp has been validated.
Write access to /var/lib/glpi/_cache has been validated.Write access to /var/lib/glpi/_cache has been validated.
Write access to /var/lib/glpi/_rss has been validated.Write access to /var/lib/glpi/_rss has been validated.
Write access to /var/lib/glpi/_uploads has been validated.Write access to /var/lib/glpi/_uploads has been validated.
Write access to /var/lib/glpi/_pictures has been validated.Write access to /var/lib/glpi/_pictures has been validated.
Write access to /var/www/html/glpi/marketplace has been validated.Write access to /var/www/html/glpi/marketplace has been validated.

**Libraries**
tmlawed/htmlawed version 1.2.5 in (/var/www/html/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.1.6 in (/var/www/html/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.5 in (/var/www/html/glpi/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/var/www/html/glpi/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/html/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/html/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/html/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/html/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/html/glpi/vendor/sabre/http/lib)
sabre/uri in (/var/www/html/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/html/glpi/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/var/www/html/glpi/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/var/www/html/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/var/www/html/glpi/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/var/www/html/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/html/glpi/vendor/sebastian/diff/src)
elvanto/litemoji in (/var/www/html/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/html/glpi/vendor/symfony/console)
scssphp/scssphp in (/var/www/html/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/html/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/html/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/html/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/html/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/html/glpi/vendor/ramsey/uuid/src)
psr/log in (/var/www/html/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/html/glpi/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/var/www/html/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/html/glpi/vendor/guzzlehttp/guzzle/src)
wapmorgan/unified-archive in (/var/www/html/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/html/glpi/vendor/paragonie/sodium_compat/src)
phpCas version 1.3.3 in (/usr/share/php)

**Plugins**
actualtime           Name: ActualTime                     Version: 1.2.0      State: Installed / not activated
	fields               Name: Additionnal fields             Version: 1.11.0     State: Enabled
	behaviors            Name: Behaviours                     Version: 2.2.2      State: Enabled
	dashboard            Name: Dashboard                      Version: 0.9.6      State: Enabled
	escalade             Name: Escalation                     Version: 2.4.4      State: Enabled
	datainjection        Name: File injection                 Version: 2.7.0      State: Installed / not activated
	formcreator          Name: Form Creator                   Version: 2.10.0     State: Enabled
	itilcategorygroups   Name: ItilCategory Groups            Version: 2.4.0      State: Enabled
	mailanalyzer         Name: Mail Analyzer                  Version: 1.6.1      State: Enabled
	mreporting           Name: More Reporting                 Version: 1.7.0      State: Enabled
	genericobject        Name: Objects management             Version: 2.9.0      State: Enabled
	tag                  Name: Tag Management                 Version: 2.7.0      State: Enabled
	tasklists            Name: Tasks list                     Version: 1.5.1      State: Enabled
	timezones            Name: Timezones                      Version: 2.5.2      State: Installed / not activated
```

### LINKED ISSUE #7649: Email import fails without image in body, succeeds with image in body but email body not added to ticket description
**Describe the bug**

When emails are imported with an IMAP receiver, the email cannot be imported unless there is an inline image in the email body.  Then, once imported, the description of the ticket is blank.  If an email is set to the mailbox where the receiver pulls from that has only text in the body, the mailgate job hangs.  Testing the receiver by telling it to 'Get email tickets now' while debug mode is enabled results in the following error:

Uncaught Exception UnexpectedValueException: is not known in /var/www/glpi/inc/mailcollector.class.php at line 2002

**Page(s) URL**

http://glpi.*****.com/front/mailcollector.form.php

**To reproduce**

Configure an IMAP receiver, pulling from an Exchange Online mailbox then:

**Test # 1**
Send an email to the account that has both an image as well as text in the body.
Force the retrieval by clicking 'Get email tickets now' on the receiver under the Actions tab.
Email should be pulled in successfully, but the ticket will not contain anything from the body of the email.

**Test # 2**
Send an email to the account that just has text in the body.
Force the retrieval by clicking 'Get email tickets now' on the receiver under the Actions tab.
The error 'Uncaught Exception UnexpectedValueException: is not known in /var/www/glpi/inc/mailcollector.class.php at line 2002' will be displayed.

**Expected behavior**

Expectation is that the body of emails pulled in by the receiver will be displayed in the description of the ticket, whether or not the body has text, images, or both.

**Logs**

No *-error.log files contain any data at the time this error occurs

**Screenshots**

None

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```

GLPI 9.5.0 ( => /var/www/glpi) Installation mode: TARBALL
--

Operating system: Linux ****.*****.com 4.18.0-193.6.3.el8_2.x86_64 #1 SMP Wed Jun 10 11:09:32 UTC 2020 x86_64 PHP 7.4.8 fpm-fcgi (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apcu, bcmath, bz2, calendar, cgi-fcgi, ctype, 	curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, gmp, hash, iconv, igbinary, imagick, imap, intl, json, ldap, libxml, 	mbstring, msgpack, mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, posix, redis, session, shmop, sockets, sodium, 	sqlite3, standard, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib) Setup: max_execution_time="600" memory_limit="128M" post_max_size="8M" safe_mode="" session.save_handler="files" 	upload_max_filesize="2M"  Software: Apache/2.4.37 (centos) () 	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.163 Safari/537.36 Server Software: MariaDB Server 	Server Version: 10.5.4-MariaDB 	Server SQL Mode:  	Parameters: glpi@localhost/glpi 	Host info: Localhost via UNIX socket 	 PHP version is at least 7.2.0 - Perfect! Sessions support is available - Perfect! Allocated memory > 64 Mio - Perfect! mysqli extension is installed ctype extension is installed fileinfo extension is installed json extension is installed mbstring extension is installed iconv extension is installed zlib extension is installed curl extension is installed gd extension is installed simplexml extension is installed intl extension is installed ldap extension is installed apcu extension is installed Zend OPcache extension is installed xmlrpc extension is installed CAS extension is installed exif extension is installed zip extension is installed bz2 extension is installed sodium extension is installed Database version seems correct (10.5.4) - Perfect! Timezones seems loaded in database The log file has been created successfully. Write access to /etc/glpi/ has been validated. Write access to /var/lib/glpi has been validated. Write access to /var/lib/glpi/_dumps has been validated. Write access to /var/lib/glpi/_sessions has been validated. Write access to /var/lib/glpi/_cron has been validated. Write access to /var/lib/glpi/_graphs has been validated. Write access to /var/lib/glpi/_lock has been validated. Write access to /var/lib/glpi/_plugins has been validated. Write access to /var/lib/glpi/_tmp has been validated. Write access to /var/lib/glpi/_cache has been validated. Write access to /var/lib/glpi/_rss has been validated. Write access to /var/lib/glpi/_uploads has been validated. Write access to /var/lib/glpi/_pictures has been validated. Write access to /var/www/glpi/marketplace has been validated. For security reasons, SELinux mode should be Enforcing.

GLPI_ROOT: /var/www/glpi GLPI_CONFIG_DIR: /etc/glpi/ GLPI_VAR_DIR: /var/lib/glpi GLPI_LOG_DIR: /var/log/glpi GLPI_MARKETPLACE_DIR: /var/www/glpi/marketplace GLPI_USE_CSRF_CHECK: 1 GLPI_CSRF_EXPIRES: 7200 GLPI_CSRF_MAX_TOKENS: 100 GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org GLPI_INSTALL_MODE: TARBALL GLPI_NETWORK_MAIL: glpi@teclib.com GLPI_NETWORK_SERVICES: https://services.glpi-network.com GLPI_MARKETPLACE_PRERELEASES:  GLPI_USER_AGENT_EXTRA_COMMENTS:  GLPI_AJAX_DASHBOARD: 1 GLPI_CALDAV_IMPORT_STATE: 0 GLPI_DEMO_MODE: 0 GLPI_FORCE_EMPTY_SQL_MODE: 1 GLPI_FORCE_NATIVE_SQL_TYPES: 1 GLPI_DOC_DIR: /var/lib/glpi GLPI_CACHE_DIR: /var/lib/glpi/_cache GLPI_CRON_DIR: /var/lib/glpi/_cron GLPI_DUMP_DIR: /var/lib/glpi/_dumps GLPI_GRAPH_DIR: /var/lib/glpi/_graphs GLPI_LOCAL_I18N_DIR: /var/lib/glpi/_locales GLPI_LOCK_DIR: /var/lib/glpi/_lock GLPI_PICTURE_DIR: /var/lib/glpi/_pictures GLPI_PLUGIN_DOC_DIR: /var/lib/glpi/_plugins GLPI_RSS_DIR: /var/lib/glpi/_rss GLPI_SESSION_DIR: /var/lib/glpi/_sessions GLPI_TMP_DIR: /var/lib/glpi/_tmp GLPI_UPLOAD_DIR: /var/lib/glpi/_uploads GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/ GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/ GLPI_I18N_DIR: /var/www/glpi/locales GLPI_VERSION: 9.5.0 GLPI_SCHEMA_VERSION: 9.5.0 GLPI_MIN_PHP: 7.2.0 GLPI_YEAR: 2020

htmlawed/htmlawed version 1.2.5 in (/var/www/glpi/vendor/htmlawed/htmlawed) phpmailer/phpmailer version 6.1.6 in (/var/www/glpi/vendor/phpmailer/phpmailer/src) simplepie/simplepie version 1.5.5 in (/var/www/glpi/vendor/simplepie/simplepie/library) tecnickcom/tcpdf version 6.3.5 in (/var/www/glpi/vendor/tecnickcom/tcpdf) michelf/php-markdown in (/var/www/glpi/vendor/michelf/php-markdown/Michelf) true/punycode in (/var/www/glpi/vendor/true/punycode/src) iamcal/lib_autolink in (/var/www/glpi/vendor/iamcal/lib_autolink) sabre/dav in (/var/www/glpi/vendor/sabre/dav/lib/DAV) sabre/http in (/var/www/glpi/vendor/sabre/http/lib) sabre/uri in (/var/www/glpi/vendor/sabre/uri/lib) sabre/vobject in (/var/www/glpi/vendor/sabre/vobject/lib) laminas/laminas-cache in (/var/www/glpi/vendor/laminas/laminas-cache/src) laminas/laminas-i18n in (/var/www/glpi/vendor/laminas/laminas-i18n/src) laminas/laminas-serializer in (/var/www/glpi/vendor/laminas/laminas-serializer/src) monolog/monolog in (/var/www/glpi/vendor/monolog/monolog/src/Monolog) sebastian/diff in (/var/www/glpi/vendor/sebastian/diff/src) elvanto/litemoji in (/var/www/glpi/vendor/elvanto/litemoji/src) symfony/console in (/var/www/glpi/vendor/symfony/console) scssphp/scssphp in (/var/www/glpi/vendor/scssphp/scssphp/src) laminas/laminas-mail in (/var/www/glpi/vendor/laminas/laminas-mail/src/Protocol) laminas/laminas-mime in (/var/www/glpi/vendor/laminas/laminas-mime/src) rlanvin/php-rrule in (/var/www/glpi/vendor/rlanvin/php-rrule/src) blueimp/jquery-file-upload in (/var/www/glpi/vendor/blueimp/jquery-file-upload/server/php) ramsey/uuid in (/var/www/glpi/vendor/ramsey/uuid/src) psr/log in (/var/www/glpi/vendor/psr/log/Psr/Log) psr/simple-cache in (/var/www/glpi/vendor/psr/simple-cache/src) mexitek/phpcolors in (/var/www/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors) guzzlehttp/guzzle in (/var/www/glpi/vendor/guzzlehttp/guzzle/src) wapmorgan/unified-archive in (/var/www/glpi/vendor/wapmorgan/unified-archive/src) paragonie/sodium_compat in (/var/www/glpi/vendor/paragonie/sodium_compat/src) phpCas version 1.3.8 in (/usr/share/pear)

Server: 'ldap://*****.*****.com', Port: '389', BaseDN: 'OU=LT Users,DC=*****,DC=com', Connection filter: 		'(&(objectClass=user)(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2))(\|(memberOf=CN=grpGLPISuperAdmins,OU=GLPI,OU=Security 		Groups,OU=LT Groups,DC=*****,DC=com)(memberOf=CN=grpGLPIAdmins,OU=GLPI,OU=Security Groups,OU=LT 		Groups,DC=*****,DC=com)))', RootDN: 'CN=svc-ldapbind,OU=Technical Services,OU=Service Accounts,OU=LT 		Users,DC=*****,DC=com', Use TLS: none

Not active

Way of sending emails: SMTP+TLS (itsupport@*****.com@smtp.office365.com)

Name: 'itsupport@*****.com' Active: Yes 	Server: '{outlook.office365.com:993/imap/ssl}INBOX' Login: 'itsupport@*****.com\itsupport' Password: Yes Name: 'itsupport@*****.com' Active: No 	Server: '{outlook.office365.com:995/pop/ssl/novalidate-cert}' Login: 'itsupport@*****.com' Password: Yes

fields               Name: Additionnal fields             Version: 1.11.0     State: Enabled 	advancedplanning     Name: advancedplanning               Version: 0.1        State: Enabled 	news                 Name: Alerts                         Version: 1.7.0      State: Enabled 	escalade             Name: Escalation                     Version: 2.5.0      State: Enabled 	formcreator          Name: Form Creator                   Version: 2.10.0     State: Enabled 	fusioninventory      Name: FusionInventory                Version: 9.5.0+1.0  State: Enabled 	uninstall            Name: Item's Lifecycle (uninstall)   Version: 2.7.0      State: Enabled 	itilcategorygroups   Name: ItilCategory Groups            Version: 2.4.0      State: Enabled 	mreporting           Name: More Reporting                 Version: 1.7.0      State: Enabled 	genericobject        Name: Objects management             Version: 2.9.0      State: Enabled 	order                Name: Orders management              Version: 2.6.0      State: Enabled


```

**Additional context**

None

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #7643 , #7649 

- Prevent exception on addresses parsing; fixes #7643
- Correctly fetch additionnal headers that may be used in rules
- Fix exclusion of mailer-daemon/postmaster
- Prevent exception when "Content-Transfer-Encoding" is not set; fixes #7649

### CODE REVIEW

File: inc/mailcollector.class.php [hunk 1]
```diff
@@ -1320,8 +1319,10 @@ function getAdditionnalHeaders(\Laminas\Mail\Storage\Message $message) {
       $head   = [];
       $headers = $message->getHeaders();
 
-      foreach ($headers as $key => $value) {
+      foreach ($headers as $header) {
          // is line with additional header?
+         $key = $header->getFieldName();
+         $value = $header->getFieldValue();
```
Comment (cedric-anne) [inline inc/mailcollector.class.php:1325]: `$key` was a numeric key here, so following `preg_match()` calls were always returning false.

File: inc/mailcollector.class.php [hunk 2]
```diff
@@ -717,9 +720,7 @@ function collect($mailgateID, $display = 0) {
                $rejinput                      = [];
                $rejinput['mailcollectors_id'] = $mailgateID;
 
-               $req_field = $this->getRequesterField();
-               $h_requester = $message->getHeader($req_field)->getAddressList();
-               $requester = $h_requester->current()->getEmail();
+               $requester = $this->getRequesterEmail($message);
```
Comment (cedric-anne) [inline inc/mailcollector.class.php:723]: `reply-to` was always defined when using the imap extension (see https://github.com/glpi-project/glpi/pull/4554#discussion_r215509231), but now this header is not available if not set in the message.

File: inc/mailcollector.class.php [hunk 3]
```diff
@@ -1354,79 +1355,71 @@ function getAdditionnalHeaders(\Laminas\Mail\Storage\Message $message) {
...
-      $h_to = $message->getHeader('to')->getAddressList();
-      $h_to->rewind();
-      $to = $h_to->current();
-
-      $reply_to_addr = null;
-      if (isset($message->reply_to)) {
-         $h_reply_to = $message->getHeader('reply_to')->getAddressList();
-         $reply_to   = $h_reply_to->current();
-         $reply_to_addr = Toolbox::strtolower($reply_to->getEmail());
+      if (preg_match('/^(mailer-daemon|postmaster)@/i', $sender_email) === 1) {
```
Comment (cedric-anne) [inline inc/mailcollector.class.php:1360]: With imap extension, we were comparing `$sender->mailbox`, which was not containing the host (i.e. `postmaster@domain.org` was returning `postmater`).
Now, we have to deal with the full email address.

File: inc/mailcollector.class.php [hunk 4]
```diff
@@ -1996,10 +2014,9 @@ public function getDecodedContent(\Laminas\Mail\Storage\Part $part) {
          case '7bit':
          case '8bit':
          case 'binary':
+         default:
```
Comment (cedric-anne) [inline inc/mailcollector.class.php:2017]: In GLPI 9.4, we were returning verbatim content if no transfer encoding was defined. I put back this behaviour as it seems that emails sent by Outlook are not containing this header for text parts (see attached email from #7649).

**Comment by dtobal** [comment]
Hi @cedric-anne sorry, but I tried this fix (inc/mailcollector.class.php) and I still have Uncaught Exception Laminas\Mail\Storage\Exception\InvalidArgumentException: Header with Name to or to not found in /var/www/html/glpi/vendor/laminas/laminas-mail/src/Storage/Part.php at line 307. Am I missing some step? I am using Office365, IMAP, SSL, NOTLS (I dont understand why NOTLS because MS says it is IMAPS TLS), and NO VALIDATE CERT. After I did upload this mailcollector... I did my connector sync ONLY ONE TIME. After that... just this error.

**Comment by trasher** [comment]
@dtobal see #7682.

**Comment by cedric-anne** [comment]
> Hi @cedric-anne sorry, but I tried this fix (inc/mailcollector.class.php) and I still have Uncaught Exception Laminas\Mail\Storage\Exception\InvalidArgumentException: Header with Name to or to not found in /var/www/html/glpi/vendor/laminas/laminas-mail/src/Storage/Part.php at line 307. Am I missing some step? I am using Office365, IMAP, SSL, NOTLS (I dont understand why NOTLS because MS says it is IMAPS TLS), and NO VALIDATE CERT. After I did upload this mailcollector... I did my connector sync ONLY ONE TIME. After that... just this error.

This one should be fixed by #7694

**Comment by dtobal** [comment]
> 
> 
> > Hi @cedric-anne sorry, but I tried this fix (inc/mailcollector.class.php) and I still have Uncaught Exception Laminas\Mail\Storage\Exception\InvalidArgumentException: Header with Name to or to not found in /var/www/html/glpi/vendor/laminas/laminas-mail/src/Storage/Part.php at line 307. Am I missing some step? I am using Office365, IMAP, SSL, NOTLS (I dont understand why NOTLS because MS says it is IMAPS TLS), and NO VALIDATE CERT. After I did upload this mailcollector... I did my connector sync ONLY ONE TIME. After that... just this error.
> 
> This one should be fixed by #7694

100% worked! Thanks!

**Pull request merged**